src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -1610,10 +1614,61 @@ public static int Sum(this IEnumerable<int> source)
             int sum = 0;
             checked
             {
-                foreach (int v in source) sum += v;
+                if (Vector.IsHardwareAccelerated && source.Count() >= SIMD_THRESHOLD)
+                    sum = VectorSum(source);
+                else
+                    foreach (int v in source) sum += v;
+            }
+            return sum;
+        }
+
+        private static int VectorSum(IEnumerable<int> source) 
+        {
+            int sum = 0;
+
+            int[] sourceArray = source.ToArray();
```

Comment:
This goes via `Buffer` which creates an internal array which might have excess zeros, and the creates yet another array if such excess does exist. Using `Buffer` directly could reduce the number of allocations needed.